vespajlib/src/main/java/com/yahoo/text/Text.java

```diff
@@ -118,19 +118,31 @@ public static OptionalInt validateTextString(String string) {
      * Validates that the given string value only contains text characters.
      */
     public static boolean isValidTextString(String string) {
-        for (int i = 0; i < string.length(); ) {
+        int length = string.length();
+        for (int i = 0; i < length; ) {
             int codePoint = string.codePointAt(i);
-            if ( ! Text.isTextCharacter(codePoint)) return false;
-
-            int charCount = Character.charCount(codePoint);
-            if (Character.isHighSurrogate(string.charAt(i))) {
-                if ( (charCount == 1) || !Character.isLowSurrogate(string.charAt(i+1))) return false;
+            if (codePoint < 0x80) {
+                if ( ! allowedAsciiChars[codePoint]) return false;
+                i++;
+            } else if (codePoint < Character.MIN_SURROGATE) {
+                i++;
+            } else {
+                if ( ! isTextCharAboveUsAscii(codePoint)) return false;
```

Comment:
Can't this return false instead of true, when the code point is out of range? 

That would save the check for isValidCodePoint.

And ... I'd assume this method is intended to do exactly what you do with the `isHighSurrogate(highSurrogate(codePoint))`?

Reply:
(That may of course not be the case ... but what is that method used for, if this isn't the case?)

Reply:
Agree that isValidCodePoint can be removed when inverting the last return value in isTextCharacter.